asmcli/lib/util.sh

```diff
@@ -647,3 +647,56 @@ get_cr_channels() {
     esac
   fi
 }
+
+ensure_cross_project_fleet_sa() {
+  local FLEET_ID; FLEET_ID="${1}"
+  local PROJECT_ID; PROJECT_ID="${2}"
+
+  if ! is_gcp; then return; fi
+
+  local FLEET_HOST_PROJECT_NUMBER
+  local FLEET_SA
+  FLEET_HOST_PROJECT_NUMBER="$(gcloud projects describe "${FLEET_ID}" --format "value(projectNumber)")"
+  FLEET_SA="service-${FLEET_HOST_PROJECT_NUMBER}@gcp-sa-gkehub.iam.gserviceaccount.com"
+
+  local FLEET_POLICIES
+  FLEET_POLICIES="$(gcloud projects get-iam-policy "${FLEET_ID}" --format=json)"
+  if ! echo "${FLEET_POLICIES}" | grep -q "gcp-sa-gkehub"; then
+    if can_register_cluster && can_modify_gcp_iam_roles; then
+      create_fleet_identity "${FLEET_ID}"
+      bind_user_to_iam_policy "roles/gkehub.serviceAgent" "${FLEET_SA}"
```

Comment:
Looks like `bind_user_to_iam_policy` used the following format and it did not fit the IAM binding for SA. 

```
gcloud projects add-iam-policy-binding "${PROJECT_ID}" \
    --member "${GCLOUD_MEMBER}" \
    --role="${role}"
```
Met error
```
2021-10-02T01:38:05.902328 asmcli: Running: '/usr/bin/gcloud projects add-iam-policy-binding test-istio-tairan --member service-634113656795@gcp-sa-gkehub.iam.gserviceaccount.com --role=roles/gkehub.serviceAgent --condition=None'
2021-10-02T01:38:05.974318 asmcli: -------------
ERROR: Policy modification failed. For a binding with condition, run "gcloud alpha iam policies lint-condition" to identify issues in condition.
ERROR: (gcloud.projects.add-iam-policy-binding) INVALID_ARGUMENT: Policy members must be of the form "<type>:<value>".
- '@type': type.googleapis.com/google.rpc.BadRequest
  fieldViolations:
  - description: Policy members must be prefixed of the form '<type>:<value>', where
      <type> is 'domain', 'group', 'serviceAccount', or 'user'.
    field: policy.bindings.member
- '@type': type.googleapis.com/google.rpc.ErrorInfo
  domain: cloudresourcemanager.googleapis.com
  reason: PROJECT_SET_IAM_DISALLOWED_MEMBER_TYPE
- '@type': type.googleapis.com/google.rpc.ResourceInfo
  resourceName: projects/test-istio-tairan

```
while SA bindings are 
```
gcloud projects add-iam-policy-binding "${FLEET_HOST_PROJECT_ID}" \
--member "serviceAccount:service-${FLEET_HOST_PROJECT_NUMBER}@gcp-sa-gkehub.iam.gserviceaccount.com" \
--role roles/gkehub.serviceAgent
gcloud projects add-iam-policy-binding "${GKE_PROJECT_ID}" \
--member "serviceAccount:service-${FLEET_HOST_PROJECT_NUMBER}@gcp-sa-gkehub.iam.gserviceaccount.com" \
--role roles/gkehub.serviceAgent
```